akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -306,6 +312,12 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
    * @param handler handler for each persisted `event`
    */
   final def persistAsync[A](event: A)(handler: A ⇒ Unit): Unit = {
+    if (insidePersistCallback) {
+      val ex = new UnsupportedOperationException("calling persistAsync from within the context of a persist/persistAsync callback is not supported!")
+      try onPersistFailure(ex, event, sequenceNr) finally context.stop(self)
```

Comment:
I need to check what actually happens when the handler throws... Supervision should kick in then?
Needs to be explicitly documented as well I think "what happens when handler throws", can add this section during this PR